crates/scarb-api/src/artifacts.rs

```diff
@@ -92,3 +92,115 @@ fn compile_artifact_at_path(path: &Utf8Path) -> Result<StarknetContractArtifacts
+                    sierra: "sierra1".to_string(),
+                    casm: "casm1".to_string(),
+                },
+                Utf8PathBuf::from("path1"),
+            ),
+        )]);
+
+        let mock_representation_1 = StarknetArtifactsRepresentation {
+            base_path: Utf8PathBuf::from("mock/path1"),
+            artifacts: StarknetArtifacts {
+                version: 1,
+                contracts: vec![StarknetContract {
+                    id: "1".to_string(),
+                    package_name: "package1".to_string(),
+                    contract_name: "contract1".to_string(),
+                    artifacts: StarknetContractArtifactPaths {
+                        sierra: Utf8PathBuf::from("mock/path1/contract1.sierra"),
+                    },
+                }],
+            },
+        };
+
+        let mock_representation_2 = StarknetArtifactsRepresentation {
+            base_path: Utf8PathBuf::from("mock/path2"),
+            artifacts: StarknetArtifacts {
+                version: 1,
+                contracts: vec![StarknetContract {
+                    id: "2".to_string(),
+                    package_name: "package2".to_string(),
+                    contract_name: "contract2".to_string(),
+                    artifacts: StarknetContractArtifactPaths {
+                        sierra: Utf8PathBuf::from("mock/path2/contract2.sierra"),
+                    },
+                }],
+            },
+        };
+
+        let representations = vec![mock_representation_1, mock_representation_2];
+
+        let result = unique_artifacts(representations, &mock_base_artifacts);
+
+        assert_eq!(result.len(), 1);
+        assert_eq!(result[0].0, "contract2");
+    }
+
+    #[test]
+    fn test_load_contracts_artifacts() {
+        let temp = crate::tests::setup_package("basic_package");
+
+        ScarbCommand::new_with_stdio()
+            .current_dir(temp.path())
+            .arg("build")
+            .run()
+            .unwrap();
+
+        ScarbCommand::new_with_stdio()
+            .current_dir(temp.path())
+            .arg("test")
+            .run()
+            .unwrap();
+
+        // Define path to the generated artifacts
+        let base_artifacts_path = temp.to_path_buf().join("target").join("dev");
+
+        // Get the base artifact
+        let base_file = Utf8PathBuf::from_path_buf(
+            base_artifacts_path.join("basic_package.starknet_artifacts.json"),
+        )
+        .unwrap();
+
+        // Load other artifact files and add them to the temporary directory
+        let other_files = vec![
+            Utf8PathBuf::from_path_buf(
+                base_artifacts_path
+                    .join("basic_package_integrationtest.test.starknet_artifacts.json"),
+            )
+            .unwrap(),
+            Utf8PathBuf::from_path_buf(
+                base_artifacts_path.join("basic_package_unittest.test.starknet_artifacts.json"),
+            )
+            .unwrap(),
+        ];
```

Comment:
It's not possible in foundry to use both `starknet_artifacts.json` and `.test.starknet_artiacts.json` files.

As a base_file, we'd use `basic_package_integrationtest.test.starknet_artifacts.json` (you can take a look at `get_starknet_artifacts_paths_from_test_targets` function, we try to get artifact with test_type `INTEGRATION_TEST_TYPE` if possible as default) and `basic_package_unittest.test.starknet_artifacts.json` as `other_files`.

Reply:
Hi @cptartur I've implemented the requested changes